docs/apis/subsystems/task/index.md

```diff
@@ -54,9 +54,18 @@ Adhoc tasks are great for situations such as:
 
 ## Usage
 
-### Failures
+### Failures and error handling
 
 A task, either scheduled or adhoc, can sometimes fail. An example would be updating an RSS field when the network is temporarily down. This is handled by the task system automatically - all the failing task needs to do is throw an exception. The task will be retried after 1 minute. If the task keeps failing, the retry algorithm will add more time between each successive attempts up to a max of 24 hours.
+Therefore, if the task fails, it is OK for your code to just throw an exception. The task system will catch it an log it for you, and schedule the retry.
+
+However, sometimes it is a good idea to catch exceptions. For example, if your task loops over courses, doing some processing for each one,
+then it is better if an exception triggered in one course does not block processing of other courses. This requires some manual exception handling.
+If you are catch exception, then you become responsible for things that the task system normally handles.
+For example, it is important to log all the details of the error, so problems can be investigated and fixed. The helper function
```

Comment:
Hi Tim,
I'm not native English but it doesn't sound good to me. Should be something like "If you catch exception"?

Reply:
Thanks for spotting this. Fixed now.